src/components/gallery/gallery.js

```diff
@@ -126,12 +127,39 @@ const groupAndSortByYear = (data) => {
 
 const TabsComponent = () => {
   const [active, setActive] = useState(1)
+  const [isModalOpen, setIsModalOpen] = useState(false)
+  const [currentImageIndex, setCurrentImageIndex] = useState(0)
+  const [modalImages, setModalImages] = useState([])
 
   const activeTabData = useMemo(() => {
     const activeTab = tabItems.find((tab) => tab.id === active)
     return activeTab ? groupAndSortByYear(activeTab.data) : []
   }, [active])
 
+  const openImageModal = (images, index, tabTitle, year) => {
+    setModalImages(images)
+    setCurrentImageIndex(index)
+    setModalTabTitle(tabTitle) // Save active tab title
+    setModalYear(year) // Save image year
+    setIsModalOpen(true)
+  }
+
+  const closeModal = () => {
+    setIsModalOpen(false)
+  }
+
+  const nextImage = () => {
+    setCurrentImageIndex((prevIndex) => (prevIndex + 1) % modalImages.length)
```

Comment:
Optional all places:
```suggestion
    setCurrentImageIndex(prevIndex => (prevIndex + 1) % modalImages.length)
```

Reply:
done